### PR TITLE
Use pytest-env to set up API key

### LINKED ISSUE #1382: Use pytest-env to set up API key shims for testing
## Problem

Presently we use a copy of the `env.template` file when testing (via the [`just dotenv`](https://github.com/WordPress/openverse-catalog/blob/a3f4a426c33909cc0f40b80f4ea5d68cd54fe7c6/justfile#L39-L41) recipe [here in the CI/CD workflow](https://github.com/WordPress/openverse-catalog/blob/8ee7fb72fbd619e568bfceaeab6e6a436386dca3/.github/workflows/ci_cd.yml#L113-L114))

The env template includes placeholders for the various API keys we use:

https://github.com/WordPress/openverse-catalog/blob/895e907b6f9594b4d43db6a07eae0b3cb4a0e7d0/env.template#L27-L34

Per @krysal's comment (https://github.com/WordPress/openverse-catalog/pull/795#discussion_r1002134568):

> Do you think we should let these variables with no value instead of the "not_set" string? 🤔 That way the scripts that require it will fail with a more clear error, before even trying to make a request, but as it's currently they take the string as the API key.

## Description

We should comment these fields out from the `env.template` file but still keep them present as examples (just like the [`env.template` files for the API](https://github.com/WordPress/openverse-api/blob/c38ddfee022fce3f72c4894a804a5577f526abeb/api/env.template)). Particularly now that we are no longer retrieving those values at DAG parse time as part of the provider refactor, we no longer need to have these defined in the environment whenever running the project. Leaving them undefined will give us more obvious errors during local DAG testing if we happen to forget to populate a variable.

We _do_ need placeholder values for testing though. That's where we might use the [pytest-env](https://pypi.org/project/pytest-env/) plugin to define these with placeholders in the [`pytest.ini` file](https://github.com/WordPress/openverse-catalog/blob/b4ef93ce5f825e4ae0e3ba3814340ec42ade7fa2/pytest.ini).

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by ngken0995**
## Fixes

Fixes #1382 by @AetherUnbound 

## Description

- comment all variables beginning with `AIRFLOW_VAR_API_KEY`
- install pytest-env
- all variables beginning with `AIRFLOW_VAR_API_KEY` are placed in `catalog/pytest.ini`

## Checklist

- [X] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [X] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [X] My commit messages follow [best practices][best_practices].
- [X] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by ngken0995** [comment]
@AetherUnbound `CI + CD / Run tests for the catalog (pull_request)` is failing. I have been trying to edit the api_key variable name and no luck on pass the test for the catalog.

**Comment by AetherUnbound** [comment]
@ngken0995 thanks for taking this on! We don't use a Pipfile for the catalog because Airflow comes with its own constraints file which we [use when building the Docker image](https://github.com/WordPress/openverse/blob/a1d0cc898f9fa1cab518b94f3b7e8b52175eb50c/catalog/Dockerfile#L68-L69).

Instead please add the `pytest-env` package (pinned to the latest version) in the `requirements_dev.txt` file:

https://github.com/WordPress/openverse/blob/d92c22a71efd4c45a1dc4ee98966a865627786db/catalog/requirements_dev.txt#L1-L12

I think the CI might be failing because the requirement is missing from that file and in the (currently unused) Pipfile instead!

**Comment by AetherUnbound** [comment]
Oh one other note, all of the `API_KEY_*` variables will need to be prefixed with `AIRFLOW_VAR_*` like they are in the `.env` file, otherwise Airflow won't recognize them as [Variables](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/variables.html) (explanation of the [environment variable piece is here](https://airflow.apache.org/docs/apache-airflow/stable/howto/variable.html#storing-variables-in-environment-variables)).

**Comment by AetherUnbound** [comment]
Update: I confirmed that with the following changes, `pytest-env` behaves as expected 😄 

```diff
diff --git a/catalog/pytest.ini b/catalog/pytest.ini
index 48029dcc4..96cafe324 100644
--- a/catalog/pytest.ini
+++ b/catalog/pytest.ini
@@ -41,14 +41,14 @@ filterwarnings=
 cache_dir=/home/airflow/.cache/pytest
 
 env =
-    API_KEY_BROOKLYN_MUSEUM=apikey
-    API_KEY_DATA_GOV=apikey
-    API_KEY_EUROPEANA=apikey
-    API_KEY_FLICKR=apikey
-    API_KEY_FREESOUND=apikey
-    API_KEY_JAMENDO=apikey
-    API_KEY_JUSTTAKEITFREE=apikey
-    API_KEY_NYPL=apikey
-    API_KEY_RAWPIXEL=apikey
-    API_KEY_THINGIVERSE=apikey
-    API_KEY_WALTERS_ART_MUSEUM=apikey
+    AIRFLOW_VAR_API_KEY_BROOKLYN_MUSEUM=apikey
+    AIRFLOW_VAR_API_KEY_DATA_GOV=apikey
+    AIRFLOW_VAR_API_KEY_EUROPEANA=apikey
+    AIRFLOW_VAR_API_KEY_FLICKR=apikey
+    AIRFLOW_VAR_API_KEY_FREESOUND=apikey
+    AIRFLOW_VAR_API_KEY_JAMENDO=apikey
+    AIRFLOW_VAR_API_KEY_JUSTTAKEITFREE=apikey
+    AIRFLOW_VAR_API_KEY_NYPL=apikey
+    AIRFLOW_VAR_API_KEY_RAWPIXEL=apikey
+    AIRFLOW_VAR_API_KEY_THINGIVERSE=apikey
+    AIRFLOW_VAR_API_KEY_WALTERS_ART_MUSEUM=apikey
diff --git a/catalog/requirements_dev.txt b/catalog/requirements_dev.txt
index d1e577fa5..00821c9c0 100644
--- a/catalog/requirements_dev.txt
+++ b/catalog/requirements_dev.txt
@@ -10,3 +10,4 @@ pytest-raises==0.11
 pytest-socket==0.6.0
 pytest-sugar==0.9.7
 pytest-xdist
+pytest-env
```

File: catalog/env.template [hunk 1]
```diff
@@ -26,17 +26,17 @@ AIRFLOW_VAR_ENVIRONMENT=dev
...
+#AIRFLOW_VAR_API_KEY_DATA_GOV=not_set
+#AIRFLOW_VAR_API_KEY_EUROPEANA=not_set
+#AIRFLOW_VAR_API_KEY_FLICKR=not_set
+#AIRFLOW_VAR_API_KEY_FREESOUND=not_set
+#AIRFLOW_VAR_API_KEY_JAMENDO=not_set
+#AIRFLOW_VAR_API_KEY_JUSTTAKEITFREE=not_set
+#AIRFLOW_VAR_API_KEY_NYPL=not_set
+#AIRFLOW_VAR_API_KEY_RAWPIXEL=not_set
+#AIRFLOW_VAR_API_KEY_THINGIVERSE=not_set
+#AIRFLOW_VAR_API_KEY_WALTERS_ART_MUSEUM=not_set
```
Comment (AetherUnbound) [inline catalog/env.template:39]: **I actually think we can do away with most of these, maybe leaving one as an example!
```suggestion
# Example: 
# AIRFLOW_VAR_API_KEY_BROOKLYN_MUSEUM=apikey
```

**Review by AetherUnbound — APPROVED** [review]
These changes look great, and runs great locally when I comment out my API keys! I have one note on the env template, otherwise this is good to go 😄

**Review by obulat — APPROVED** [review]
Awesome, nice to simplify the variable creation, @ngken0995!

**Pull request merged**